src/System.Private.ServiceModel/src/Internals/System/Runtime/ServiceModelSynchronizationContext.cs

```diff
@@ -12,7 +12,7 @@ public class ServiceModelSynchronizationContext : SynchronizationContext
 
         public override void Post(SendOrPostCallback d, object state)
         {
-            IOThreadScheduler.ScheduleCallbackNoFlow(d, state);
+            IOThreadScheduler.ScheduleCallbackNoFlow((s) => d(s), state);
```

Comment:
This is going to allocate a closure and a delegate on every invocation. Is it called low-frequency enough that the overhead doesn't matter?